docs/configuration.rst

```diff
@@ -472,3 +477,69 @@ Optional Settings
+
+.. sourcecode:: bash
+
+  docker run -d \
+    -p 127.0.0.1:8088:8088
+    -e KSQL_BOOTSTRAP_SERVERS=localhost:9092 \
+    -e KSQL_LISTENERS=http://0.0.0.0:8088 \
+    -e KSQL_KSQL_SERVICE_ID=ksql_test_service_id_ \
+    -e KSQL_KSQL_STREAMS_REPLICATION_FACTOR=3 \
+    -e KSQL_KSQL_SINK_REPLICAS=3 \
+    -e KSQL_KSQL_STREAMS_STATE_DIR=/docker/container/path \
+    -v /docker/host/path:/docker/container/path \
+    confluentinc/cp-ksql-server:4.1.2
+
+Docker Options
+""""""""""""""
+
+* Data persistence: KSQL Server runs Kafka Streams applications which may need to store persistent state locally. The command above maps the state store directory inside the container (cf. KSQL_KSQL_STREAMS_STATE_DIR) to a directory on the Docker host.
+* Port mapping: The above command maps the listener port to 8088 on the Docker host, hence ensuring that other containers (like for the KSQL CLI) running on the same host can connect to the KSQL server container.
+
+Required Settings
+"""""""""""""""""
+The following settings must be passed to run the KSQL Server image.
+
+``KSQL_BOOTSTRAP_SERVERS``
+
+To configure the Kafka cluster that KSQL should read from/write to, you must specify one or more "bootstrap" brokers from that Kafka cluster. This parameter is formatted as a comma-separated list of hostname:port pairs; for example, broker-hostname1:9092,broker-hostname2:9092. These brokers are used by KSQL to establish the initial connection to the Kafka cluster and to discover the set of all available brokers in the Kafka cluster, the set of which may change dynamically at runtime (e.g. brokers may be taken down for maintenance). You can but don't need to specify all brokers in the Kafka cluster, though it is recommended to specify at least two brokers to make bootstrapping more resilient against individual broker outages.
+
+Optional Settings
+"""""""""""""""""
+All other settings for KSQL like :ref:`security <ksql-security>`, monitoring interceptors, overrides for Kakfa Streams, Kafka Producer, or Kakfa Consumer settings can be passed to the Docker images as environment variables. The names of these environment variables are derived by prefixing with KSQL_, uppercasing each word in the parameter name, and separating each word with an underscore (_) instead of periods (.). For example, if you need to set ``ksql.streams.state.dir``, the environment variable name would be ``KSQL_KSQL_STREAMS_STATE_DIR``.
+
+The image will then convert these environment variables to corresponding KSQL Server config variables.
+
+--------
+KSQL CLI
+--------
+The KSQL CLI image will execute the ``ksql`` command and try to connect to the server at ``http://localhost:8088`` by default, as shown below. 
+
+.. sourcecode:: bash
+
+  docker run -it -rm confluentinc/cp-ksql-cli:4.1.2
+
+Note: Because the CLI is interactive, the container cannot log to STDOUT as is customary for Docker containers. The KSQL CLI logs can instead be found at /var/logs/ksql-cli/ inside the container.
```

Comment:
Can one still use `docker log ...` to see the STDOUT logs?

Reply:
See my response here: https://github.com/confluentinc/cp-docker-images/pull/496#discussion_r191879567